jina/peapods/zmq/__init__.py

```diff
@@ -785,7 +788,8 @@ def _parse_from_frames(sock_type, frames: List[bytes]) -> 'Message':
         frames = [b' '] + frames
     elif sock_type == zmq.ROUTER:
         # the router appends dealer id when receive it, we need to remove it
-        frames.pop(0)
+        if len(frames) == 4:
```

Comment:
what's the reason for this?

Reply:
If we have `ROUTER-ROUTER` communication, there are just three frames. Only in `DEALER-ROUTER` communication we have to pop the first frame.